### DISCUSSION TITLE
The purpose of [...]posed might be totally irrelevant? in the future.

### DISCUSSION THREAD

**Discussion by eebssk1** [LSPosed/LSPosed#2850] [General]
Nowadays more and more apps are using google play automatic intergrity protection (know as libpairipcore.so) to protect their apps.

Lsposed currently does not work on apps(in the module scope) with it. As it causes invalid memory access error when start app.

The discuss is started to let people talk about possible solution on it.
Maybe someday it will becomes the default and lsposed totally lose the purpose(though system module won't affect.)

For more information, check https://github.com/rednaga/APKiD/issues/329

**Comment by mywalkb** [comment]
> For more information, check [rednaga/APKiD#329](https://github.com/rednaga/APKiD/issues/329)

In that issue there is [this](https://github.com/rednaga/APKiD/issues/329#issuecomment-1809621534) comment, which reference a solution proposed by NotAWeeb! in a external forum. Just don't instance pairip Application but original. So you can set an hook on constructor and return an instance of android.app.Application.
Have you some apps to test my theory?

**Reply by eebssk1** [reply]
Yes this is also my first solution I come up with, I will find some time to test.

It should work in theory as long as the games does not load outer  encrypted codes.

**Reply by eebssk1** [reply]
So  I tried. Only a few apps work without pairipcore. A lot need this library on startup to restore random encrypted strings.
